Source/ARTDeviceDetails.m

```diff
@@ -0,0 +1,50 @@
+//
+//  ARTDeviceDetails.m
+//  Ably
+//
+//  Created by Ricardo Pereira on 07/02/2017.
+//  Copyright © 2017 Ably. All rights reserved.
+//
+
+#import "ARTDeviceDetails.h"
+#import "ARTDevicePushDetails.h"
+
+NSString *const ARTDevicePlatform = @"ios";
+
+NSString *ARTDeviceFormFactorToStr(ARTDeviceFormFactor formFactor) {
+    switch (formFactor) {
+        case ARTDeviceFormFactorMobile:
+            return @"mobile"; //0
+        case ARTDeviceFormFactorTablet:
+            return @"tablet"; //1
+        case ARTDeviceFormFactorDesktop:
+            return @"desktop"; //2
+        case ARTDeviceFormFactorEmbedded:
+            return @"embedded"; //3
+    }
+}
+
+@implementation ARTDeviceDetails
+
++ (instancetype)fromLocalDevice {
+    return [[ARTDeviceDetails alloc] init];
+}
+
+- (instancetype)init {
+    return [self initWithId:[[NSUUID new] UUIDString]];
```

Comment:
> I will look into the paper of ULID and try to implement it.

I don't think we should be writing our own implementation if we can help it.  Is that really necessary if we can't find a lib that does it for us?